libpod/networking_pasta_linux.go

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// networking_pasta_linux.go - Start pasta(1) for user-mode connectivity
+//
+// Copyright (c) 2022 Red Hat GmbH
+// Author: Stefano Brivio <sbrivio@redhat.com>
+
+package libpod
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/containernetworking/plugins/pkg/ns"
+	"github.com/sirupsen/logrus"
+)
+
+func (r *Runtime) setupPasta(ctr *Container, netns ns.NetNS) error {
+	var NoTCPInitPorts = true
+	var NoUDPInitPorts = true
+	var NoTCPNamespacePorts = true
+	var NoUDPNamespacePorts = true
+
+	path := r.config.Engine.NetworkCmdPath
+	if path == "" {
+		var err error
+		path, err = exec.LookPath("pasta")
+		if err != nil {
+			return fmt.Errorf("could not find pasta, the network namespace can't be configured: %w", err)
+		}
+	}
+
+	cmdArgs := []string{}
+	cmdArgs = append(cmdArgs, "--config-net")
+
+	for _, i := range ctr.convertPortMappings() {
+		protocols := strings.Split(i.Protocol, ",")
+		for _, protocol := range protocols {
+			var addr string
+
+			if i.HostIP != "" {
+				addr = fmt.Sprintf("%s/", i.HostIP)
+			} else {
+				addr = ""
+			}
```

Comment:
not needed a string in go is by default initialized as empty

Reply:
Dropped.